### PR TITLE
Move most of the TOPP test for DatabaseSuitability to THIRDPARTY

### LINKED ISSUE #5907: Deactivate DatabaseSuitability_test by default
Only activate on our infrastructure where we have novoR.
We basically need a CMake option for that, like for CSIFingerID.

### PR DESCRIPTION
**Pull request by Waschi97**
# Description

Since DatabaseSuitability internally performs an ID search by calling CometAdapter the tests that do that were moved to THIRDPARTY.
Also added a new DatabaseSuitability test to the standard TOPP tests that uses the `-algorithm:disable_correction` flag which turns off the dependency on Comet.

Closes #5907

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.
- Or click on the action: e.g., for clang-format linting

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

# Advanced commands (admins / reviewer only):
- `/rebase` will try to rebase the PR on the current develop branch.
- `/reformat` (experimental) applies the clang-format style changes as additional commit
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds

### CODE REVIEW

**Review by timosachsenberg — APPROVED** [review]

**Comment by jpfeuffer** [comment]
Can you move it here instead?
https://github.com/OpenMS/OpenMS/blob/23e499e5fcbe10c513e03fc26b59721b3d06a711/src/tests/topp/THIRDPARTY/third_party_tests.cmake#L178

**Review by jpfeuffer — CHANGES_REQUESTED** [review]
see my comment

**Comment by Waschi97** [comment]
I thought executing the DatabaseSuitability tests after the CometAdapter tests made sense since DatabaseSuitability depends on it.
But I guess even if the CometAdapter tests fail this wouldn't stop the DatabaseSuitability tests to be performed... So yeah, sure I can move it up.

**Comment by jpfeuffer** [comment]
It is not really up or down. It is a different file and the most important thing is that the existence of a comet executable is checked first.
You can of course move it to the end of the if block

**Comment by Waschi97** [comment]
wait... but it is inside of the if block

**Comment by jpfeuffer** [comment]
aaah.. sorry I was confused by the second part of the tests that is still in the normal cmake.

File: src/tests/topp/CMakeLists.txt [hunk 1]
```diff
@@ -226,18 +226,11 @@ set_tests_properties("TOPP_MapNormalizer_1_out1" PROPERTIES DEPENDS "TOPP_MapNor
...
-add_test("TOPP_DatabaseSuitability_2_out" ${DIFF} -whitelist ${INDEX_WHITELIST} -in1 DatabaseSuitability_2.tmp -in2 ${DATA_DIR_TOPP}/DatabaseSuitability_out_2.tsv )
-set_tests_properties("TOPP_DatabaseSuitability_2_out" PROPERTIES DEPENDS "TOPP_DatabaseSuitability_2")
-# test with custom FDR
-add_test("TOPP_DatabaseSuitability_3" ${TOPP_BIN_PATH}/DatabaseSuitability -test -in_id ${DATA_DIR_TOPP}/DatabaseSuitability_in_id.idXML -in_spec ${DATA_DIR_TOPP}/DatabaseSuitability_in_spec.mzML -in_novo ${DATA_DIR_TOPP}/DatabaseSuitability_in_novo.idXML -database ${DATA_DIR_TOPP}/DatabaseSuitability_database.fasta -novo_database ${DATA_DIR_TOPP}/DatabaseSuitability_novo_database.FASTA -algorithm:FDR 0.5 -algorithm:reranking_cutoff_percentile 0.5 -out DatabaseSuitability_3.tmp)
-add_test("TOPP_DatabaseSuitability_3_out" ${DIFF} -whitelist ${INDEX_WHITELIST} -in1 DatabaseSuitability_3.tmp -in2 ${DATA_DIR_TOPP}/DatabaseSuitability_out_3.tsv )
-set_tests_properties("TOPP_DatabaseSuitability_3_out" PROPERTIES DEPENDS "TOPP_DatabaseSuitability_3")
+# test with diasabled correction (not depending on Comet)
```
Comment (jpfeuffer) [inline src/tests/topp/CMakeLists.txt:229]:
```suggestion
# test with disabled correction (not depending on Comet)
```

**Review by jpfeuffer — APPROVED** [review]

**Pull request merged**